### PR TITLE
Add io.github.grangerhub.Tremulous

### PR DESCRIPTION
**Pull request by AsciiWolf**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
https://github.com/GrangerHub/tremulous/pull/130

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

This is just a rename of [com.grangerhub.Tremulous](https://github.com/flathub/com.grangerhub.Tremulous) because the original domain has been lost. There are no other changes.

Tremulous is a free, open source game that blends a team based FPS with elements of an RTS. It is based on a modified Quake 3 engine.

Please, also send an invite to @cengique (the upstream developer) when this PR gets accepted.

### CODE REVIEW

**Comment by AsciiWolf** [comment]
bot, build

**Comment by AsciiWolf** [comment]
bot, build io.github.grangerhub.Tremulous

**Comment by hfiguiere** [comment]
why weren't the appstream, icon and desktop file never submitted upstream?

**Comment by hfiguiere** [comment]
"not suitable" isn't an excuse as they are generic to XDG support and not flatpak specific.

**Comment by hfiguiere** [comment]
Also

> Please describe your application briefly.

(say it a replacement for an undescript app isn't helpful)

**Comment by cengique** [comment]
Tested, running fine. Except I have a sound issue that may be related to my test environment:
```
------ Initializing Sound ------
SDL_Init( SDL_INIT_AUDIO )... FAILED (Pipewire: Failed to connect hotplug detection context (112))
```

> why were the appstream, icon and desktop file never submitted upstream?

I can submit these upstream if needed.

**Comment by cengique** [comment]
Also, the audio issue was due to specific system tested - it works fine on another system.

File: io.github.grangerhub.Tremulous.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  <url type="homepage">https://github.com/GrangerHub/tremulous</url>
+  <screenshots>
+    <screenshot type="default"><image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/tremulous/a.png</image></screenshot>
+    <screenshot><image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/tremulous/b.png</image></screenshot>
+    <screenshot><image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/tremulous/c.png</image></screenshot>
+    <screenshot><image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/tremulous/d.png</image></screenshot>
+    <screenshot><image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/tremulous/e.png</image></screenshot>
+  </screenshots>
+  <update_contact>mail@asciiwolf.com</update_contact>
+  <content_rating type="oars-1.0">
```
Comment (bbhtt) [inline io.github.grangerhub.Tremulous.appdata.xml:34]:
```suggestion
  <content_rating type="oars-1.1">
```

Then you can remove all the none values

**Comment by bbhtt** [comment]
> I can submit these upstream if needed.

Yes please do.

**Comment by AsciiWolf** [comment]
> I can submit these upstream if needed.

Thanks! Make a Pull request for this and let me know. I will help with any changes/fixed needed.

File: io.github.grangerhub.Tremulous.appdata.xml (hunk 1 above)
Comment (AsciiWolf) [reply io.github.grangerhub.Tremulous.appdata.xml]: Done, thanks! :)

**Comment by AsciiWolf** [comment]
bot, build io.github.grangerhub.Tremulous

**Comment by bbhtt** [comment]
> How does this differ from [com.grangerhub.Tremulous](https://flathub.org/apps/com.grangerhub.Tremulous)?

> This is just a rename of [com.grangerhub.Tremulous](https://github.com/flathub/com.grangerhub.Tremulous) because the original domain has been lost. There are no other changes.

I forgot to tag it as migrate.

**Comment by AsciiWolf** [comment]
@cengique Let me know if you need any help with the Pull request adding desktop / AppStream files. I would prepare the PR myself, sadly my Makefile editing skills are (still) not ideal.

**Comment by cengique** [comment]
> @cengique Let me know if you need any help with the Pull request adding desktop / AppStream files. I would prepare the PR myself, sadly my Makefile editing skills are (still) not ideal.

Thanks @AsciiWolf . I have not forgotten about this, but have been temporarily occupied. I was also a bit lost, so let me run it by you. I plan to create a new folder on the upstream repo to put the three files (appstream, icon, desktop, and also copy the screenshots from the fedora repo they come from) in there. The Makefile patch is really rudimentary - it can even be left out, but I'll try to patch it.

**Comment by AsciiWolf** [comment]
No worries. You don't have to include the Makefile patch as it is a Flatpak-specific thing. The second patch as far as I know is already in master. The screenshots from Fedora do not look ideal and I originally made better ones for my Tremulous Snap, but forgot to use them for the Flatpak and have already deleted the original files (and the compressed Snap Store ones look awful). Maybe we could make another.

Anyway, what I was talking about was adding the AppStream and desktop/icon file install rules to the Makefile. That's something I would rather leave to someone more skilled.

**Comment by cengique** [comment]
> No worries. You don't have to include the Makefile patch as it is a Flatpak-specific thing. The second patch as far as I know is already in master. The screenshots from Fedora do not look ideal and I originally made better ones for my Tremulous Snap, but forgot to use them for the Flatpak and have already deleted the original files (and the compressed Snap Store ones look awful). Maybe we could make another.

I'm working on a pull request. I found some nice screenshots and also archived the ones I found in different places. 

> Anyway, what I was talking about was adding the AppStream and desktop/icon file install rules to the Makefile. That's something I would rather leave to someone more skilled.

Ok, I'm clueless on that as well, however, I just did some reading. The main issue is that our Makefile does not even have an "install" section. The install scripts are per distro. Here's an example from Arch linux: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=tremulous-grangerhub-bin
Potentially I can bring that into our Makefile and put our appstream file under `/usr/share/metainfo/`. At some point, I remember someone implementing Cmake support as well, but I would have to do some more research for that as it's giving me an error currently.

**Comment by AsciiWolf** [comment]
bot, build io.github.grangerhub.Tremulous

**Comment by AsciiWolf** [comment]
bot, build io.github.grangerhub.Tremulous

**Comment by AsciiWolf** [comment]
> > I can submit these upstream if needed.
> 
> Yes please do.

https://github.com/GrangerHub/tremulous/pull/130

**Comment by AsciiWolf** [comment]
bot, build io.github.grangerhub.Tremulous

**Comment by AsciiWolf** [comment]
screenshot

I forgot to mention that it was tested and works fine. ;-)

**Comment by AsciiWolf** [comment]
@barthalion Please also invite @cengique

Thanks! :-)

**Comment by AsciiWolf** [comment]
@barthalion @cengique The stable build seems to fail with "appid-code-host-not-reachable". Any idea?

https://buildbot.flathub.org/#/builders/6/builds/131497

**Comment by bbhtt** [comment]
We're investigating, please wait.

**Comment by cengique** [comment]
@barthalion @bbhtt Can you add me back to https://github.com/flathub/io.github.grangerhub.Tremulous and to the developer portal for the flatpak?

**Comment by bbhtt** [comment]
> @barthalion @bbhtt Can you add me back to https://github.com/flathub/io.github.grangerhub.Tremulous and to the developer portal for the flatpak?

Added you

**Pull request closed without merging**